src/server/general/creation.md

```diff
@@ -0,0 +1,99 @@
+# Creating the Display and event loops
+
+wayland-server provides a way to instantiate a `Display` for use by the compositor. Simply calling the `new` function creates a display. However this display is not exposed to clients. To allow clients to connect to the `Display`, a socket needs to be created. A socket may be created through the [`Display::add_socket_auto`] which creates a socket using the first available name.
+
+For compositors which wish to have more control, you may also specify the name of the socket to use with the [`Display::add_socket`], [`Display::add_socket_fd`] or [`Display::add_socket_from`] functions.
+
+```rust,norun
+use std::env;
+use wayland_server::Display;
+
+let display = Display::new();
+
+let socket_name = display
+    .add_socket_auto()
+    .expect("Failed to add wayland socket")
+    .into_string()
+    .unwrap();
+
+println!("Listening on wayland socket {}", socket_name);
+env::set_var("WAYLAND_DISPLAY", &socket_name);
+```
+
+At this point a client can attempt to connect to the compositor. However you may notice using [wayland-info](https://gitlab.freedesktop.org/wayland/wayland-utils) that when running wayland-info, the application hangs with no response. The cause of this behavior is that the compositor's `Display` does not respond to any requests from the client and wayland-info will wait until it receives a response.
```

Comment:
I believe this is the case the compositor doesn't read the messages, the flush is what replies to clients below?